packages/apidom-ls/src/config/common/schema/lint/missing-core-fields-oas-3-0-asyncapi.ts

```diff
@@ -0,0 +1,91 @@
+import { DiagnosticSeverity } from 'vscode-languageserver-types';
+
+import ApilintCodes from '../../../codes';
+import { LinterMeta } from '../../../../apidom-language-types';
+import { OpenAPI3 } from '../../../openapi/target-specs';
+import { AsyncAPI } from '../../../asyncapi/target-specs';
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+const missingCoreFieldsOpenAPI3_0AndAsyncAPILint: LinterMeta = {
```

Comment:
IMHO this needs to be forked to multipe rules targeting different specs.

OpenAPI 2.0/3.0.x uses JSON Schema Draft 4/5 with following keywords: https://github.com/swagger-api/apidom/blob/main/packages/apidom-ns-json-schema-draft-4/src/elements/JSONSchema.ts#L35

AsyncAPI 2.x uses JSON Schema Draft 7 which:
Removes keywords: id
Add keywords: $id, contains, propertyNames, examples, $comment, if, then, else, contentEncoding, contentMediaType, media, writeOnly

OpenAPI 3.1 uses JSON Schema Draft 2020-12 with following keywords: https://github.com/swagger-api/apidom/blob/5b504e523c532a31aaf1b133a14ff3b76c126435/packages/apidom-ns-openapi-3-1/src/elements/Schema.ts